arangod/Replication2/ReplicatedLog/LogCommon.cpp

```diff
@@ -500,6 +511,7 @@ auto replicated_log::to_string(CommitFailReason const& r) -> std::string {
 
 auto replicated_log::operator==(CommitFailReason const& left,
                                 CommitFailReason const& right) noexcept -> bool {
+  // Ignoring the `who` field so that we only compare the reason of failure.
   return left.value.index() == right.value.index();
```

Comment:
What is the implication of that? I don't remember what the `operator==` is used for, but it might be confusing it something is not updated, because the index did not change, although the `who` changed. I imagen a situation like: `Can not commit because of X` - you fix `X` - now you can not commit of `Y` but the message still say `X`.

Reply:
We discussed this and my reasoning (which I don't claim is sound) was: 

The `who` is only informational, the error is that quorum size wasn't reached; (and I don't want to deal with `std::variant` right now)

It is certainly possible (and in view of what you say desirable) to compare `who`.

Reply:
Let me just mention, that `std::variant` forward comparison [operators](https://en.cppreference.com/w/cpp/utility/variant/operator_cmp). You only have to define an `operator==` for each type of the variant.

And then I think it could even be a space shit.

Reply:
Declared `operator==` as `default` for each type of the variant and for `CommitFailReason`. The `who` member will be taken into account.